sklearn/ensemble/_weight_boosting.py

```diff
@@ -138,6 +148,28 @@ def fit(self, X, y, sample_weight=None):
         # Check parameters
         self._validate_estimator()
 
+        if is_regressor(self):
+            routed_params = process_routing(
+                obj=self, method="fit", other_params=fit_params
+            )
```

Comment:
right now AdaBoostRegressor doesn't pass sample_weight to base_estimator even if the user explicitly passes them. Should this behavior be changed?

Reply:
If `base_estimator` is configured to require `sample_weight`, I think we pass it in. If it is not configured, we do not pass it in, which is consistent with the current behavior.

Reply:
Thinking it over, I think  want to go with a more lenient approach: https://github.com/scikit-learn/scikit-learn/pull/24026#discussion_r957504426

Reply:
But currently `AdaBoostRegressor` never passes `sample_weight` to the `base_estimator_`.

Reply:
`AdaBoostRegressor` currently does not pass in `sample_weight`, but if `base_estimator` requested it I think `sample_weight` should be passed in.

If `AdaBoostRegressor` does not pass in `sample_weight` to `base_estimator` when requested, then the API seems inconsistent. A router can ignore metadata requested from the `base_estimator`.